parse_onet_mapping.py

```diff
@@ -0,0 +1,129 @@
+from bs4 import BeautifulSoup
+import pymysql
+import openpyxl
+import requests
+import base64
+import math
+from concurrent.futures import ThreadPoolExecutor, as_completed
+
+from manager import executeQuery
+
+# Executes any Select Query
+def executeSelect(query):
+    connection = pymysql.connect(
+                user     = "group5a",
+                password = "_}2nSW6%?3hyr9Z",
+                host     = "mysql.labthreesixfive.com",
+                db       = "group5a",
+                port     = 3306
+    )
+    with connection.cursor() as cursor:
+        cursor.execute(query)
+    connection.commit()
+    return cursor.fetchall()
+
+# Gets ONET descriptor codes for each job descriptor
+def get_characteristic_codes():
+    path = './onet_mapping.xlsx'
+ 
+    wb_obj = openpyxl.load_workbook(path) # To open the workbook, workbook object is created
+    sheet_obj = wb_obj.active # Get workbook active sheet object from the active attribute
+    
+    max_col = sheet_obj.max_column
+    max_row = sheet_obj.max_row
+    profile_characteristic_codes = {}
+    for row in range(7, max_row + 1): # skip column titles
+        
+        descriptor_codes = []
+        profile_characteristic = ''
+        for col in range(1, max_col + 1):
+            cell_obj = sheet_obj.cell(row = row, column = col)
+
+            if col == 1:
+                profile_characteristic = cell_obj.value
+
+            if cell_obj.hyperlink:
+
+                # Get ONET descriptor code from link
+                descriptor_link = cell_obj.hyperlink.target
+                descriptor_code = descriptor_link.split("/")[-1]
+                if descriptor_code:
+                    descriptor_codes.append(descriptor_code)
+        
+        if descriptor_codes:
+
+            # Last character is the profile characteristic id
+            profile_characteristic_codes[profile_characteristic.split()[-1]] = descriptor_codes
+    return profile_characteristic_codes
+
+# Selects all profile ids and job codes from ONET Profile
+def get_onet_profiles():
+    query = """SELECT Id, Code FROM Profiles WHERE Category = 'ONET' """
+    ids, codes = zip(*executeSelect(query))
+    return ids, codes
+
+def fetch_onet_job(profile_id, job_code, profile_characteristic_codes):
+    queries = []
+
+    # Pinging ONET API 
+    api_endpoint = f"https://services.onetcenter.org/ws/online/occupations/{job_code}/details/"
+    username = 'raythked'
+    password =  '5638nqd'
+    headers = {
+        'Authorization': 'Basic ' + base64.standard_b64encode((username + ':' + password).encode()).decode(),
+    }
+
+    response = requests.get(api_endpoint, headers=headers)
+    soup = BeautifulSoup(response.content, "lxml")
```

Comment:
Using beautiful soup to look up in xml file

Reply:
was parsing xml necessary or your choice? on my end with JS, there was options to get json files returned. If this approach works I have no complaints though.